apps/opencs/view/render/terrainshapemode.cpp

```diff
@@ -0,0 +1,1459 @@
+    {
+        std::string cellId = CSMWorld::CellCoordinates::generateId(cellCoordinates.getX(), cellCoordinates.getY());
+        const CSMWorld::LandHeightsColumn::DataType landShapePointer = landTable.data(landTable.getModelIndex(cellId, landshapeColumn)).value<CSMWorld::LandHeightsColumn::DataType>();
+        const CSMWorld::LandHeightsColumn::DataType landRightShapePointer = landTable.data(landTable.getModelIndex(CSMWorld::CellCoordinates::generateId(cellCoordinates.getX() + 1, cellCoordinates.getY()), landshapeColumn)).value<CSMWorld::LandHeightsColumn::DataType>();
+        const CSMWorld::LandHeightsColumn::DataType landDownShapePointer = landTable.data(landTable.getModelIndex(CSMWorld::CellCoordinates::generateId(cellCoordinates.getX(), cellCoordinates.getY() + 1), landshapeColumn)).value<CSMWorld::LandHeightsColumn::DataType>();
+        const CSMWorld::LandNormalsColumn::DataType landNormalsPointer = landTable.data(landTable.getModelIndex(cellId, landnormalsColumn)).value<CSMWorld::LandNormalsColumn::DataType>();
+        CSMWorld::LandNormalsColumn::DataType landNormalsNew(landNormalsPointer);
+
+        // Generate land normals record
+        for(int i = 0; i < ESM::Land::LAND_SIZE; ++i)
+        {
+            for(int j = 0; j < ESM::Land::LAND_SIZE; ++j)
+            {
+                osg::Vec3f v1;
+                osg::Vec3f v2;
+                osg::Vec3f normal;
+                float hyp;
+
+                v1.x() = 128;
+                v1.y() = 0;
+                if (i < ESM::Land::LAND_SIZE - 1) v1.z() = landShapePointer[j * ESM::Land::LAND_SIZE + i + 1] - landShapePointer[j * ESM::Land::LAND_SIZE + i];
+                else
+                {
+                    std::string cellId = CSMWorld::CellCoordinates::generateId(cellCoordinates.getX() + 1, cellCoordinates.getY());
+                    if (!noCell(cellId) && !noLand(cellId) && noLandLoaded(cellId))
+                        v1.z() = landRightShapePointer[j * ESM::Land::LAND_SIZE + 1] - landShapePointer[j * ESM::Land::LAND_SIZE + i];
+                    else
+                        v1.z() = 0;
+                }
+
+                v2.x() = 0;
+                v2.y() = 128;
+                if (j < ESM::Land::LAND_SIZE - 1) v2.z() = landShapePointer[(j + 1) * ESM::Land::LAND_SIZE + i] - landShapePointer[j * ESM::Land::LAND_SIZE + i];
+                else
+                {
+                    std::string cellId = CSMWorld::CellCoordinates::generateId(cellCoordinates.getX(), cellCoordinates.getY() + 1);
+                    if (!noCell(cellId) && !noLand(cellId) && noLandLoaded(cellId))
+                        v2.z() = landDownShapePointer[ESM::Land::LAND_SIZE + i] - landShapePointer[j * ESM::Land::LAND_SIZE + i];
+                    else
+                        v2.z() = 0;
+                }
+
+                normal = v1 ^ v2;
+
+                hyp = normal.length() / 127.0f;
+
+                normal /= hyp;
```

Comment:
Declare and initialize variables at the same time:
```c++
osg::Vec3f normal = v1 ^ v2;
const float hyp = normal.length() / 127.0f;
```

Reply:
Just good coding habit I assume? Also, `const` works, but I didn't think it's a big deal here.